test/extensions/filters/udp/dns_filter/dns_filter_test_utils.h

```diff
@@ -17,6 +17,112 @@ std::string buildQueryForDomain(const std::string& name, uint16_t rec_type, uint
 void verifyAddress(const std::list<std::string>& addresses, const DnsAnswerRecordPtr& answer);
 size_t getResponseQueryCount(DnsMessageParser& parser);
 
+/**
+ * @brief Extracts the protocol name from the fully qualified service name. The leading underscore
+ * is discarded from the output
+ */
+absl::string_view getProtoFromName(const absl::string_view name);
+
+class DnsResponseValidator : public Logger::Loggable<Logger::Id::filter> {
+public:
+  struct DnsAnswerCtx {
+    DnsAnswerCtx(const Buffer::InstancePtr& buffer, const absl::string_view record_name,
+                 const uint16_t record_type, const uint16_t record_class,
+                 const uint16_t available_bytes, const uint16_t data_length, const uint32_t ttl,
+                 uint64_t& offset)
+        : buffer_(buffer), record_name_(record_name), record_type_(record_type),
+          record_class_(record_class), available_bytes_(available_bytes), data_length_(data_length),
+          ttl_(ttl), offset_(offset) {}
+
+    const Buffer::InstancePtr& buffer_;
+    const std::string record_name_;
+    const uint16_t record_type_;
+    const uint16_t record_class_;
+    const uint16_t available_bytes_;
+    const uint16_t data_length_;
+    const uint32_t ttl_;
+    uint64_t& offset_;
+  };
+
+  /**
+   * @brief Parse the response DNS message and create a context object
+   *
+   * @param client_request a structure containing addressing information and the buffer received
+   * from a client
+   */
+  DnsQueryContextPtr createResponseContext(Network::UdpRecvData& client_request,
```

Comment:
All these methods could just be made static, as this class doesn't have any member variables and doesn't hold state.

Reply:
Make sense, let me make them static.